### PR TITLE
UCT/GDRCOPY: Require rcache by default

### PR DESCRIPTION
**Pull request by brminich**
## What
Require rcache to be used by gdr copy md by default

## Why ?
Otherwise there maybe an issue where small memory region to be registered is aligned to 64K and IB registration fails with the following errors
```
ib_md.c:276  UCX  ERROR ibv_reg_dmabuf_mr(address=0x7f271a600000, length=65536, access=0xf) failed: Invalid argument
ucp_mm.c:70   UCX  ERROR failed to register address 0x7f271a600000 (cuda) length 65536 dmabuf_fd 98 on md[2]=mlx5_0: Input/output error (md supports: host|cuda)
```
## How
 Keep `UCX_GDR_COPY_RCACHE` env var, but make it bool instead of ternary. So it is always enabled by default, but there is still a possibility to debug/use gdrcopy transport without rcache

### CODE REVIEW

**Review by Artemy-Mellanox — APPROVED** [review]

**Review by tvegas1 — APPROVED** [review]

**Review by ivankochin — DISMISSED** [review]

File: src/uct/cuda/gdr_copy/gdr_copy_md.c [hunk 1]
```diff
@@ -42,8 +42,8 @@ static ucs_config_field_t uct_gdr_copy_md_config_table[] = {
      "shared across all contexts",
      ucs_offsetof(uct_gdr_copy_md_config_t, shared), UCS_CONFIG_TYPE_BOOL},
 
-    {"RCACHE", "try", "Enable using memory registration cache",
-     ucs_offsetof(uct_gdr_copy_md_config_t, enable_rcache), UCS_CONFIG_TYPE_TERNARY},
+    {"RCACHE", "yes", "Enable using memory registration cache",
+     ucs_offsetof(uct_gdr_copy_md_config_t, enable_rcache), UCS_CONFIG_TYPE_BOOL},
```
Comment (rakhmets) [inline src/uct/cuda/gdr_copy/gdr_copy_md.c:46]: Is it safe to change the option type from `ternary` to `bool`? How do we handle the case when the new version will be run with `RCACHE=try`? If there is a crash, then it is better to keep it `ternary`, but print a message that `try` is treated as `yes`, and will be removed.

Comment (Artemy-Mellanox) [reply src/uct/cuda/gdr_copy/gdr_copy_md.c:46]: we haven't released a version with a ternary variant yet, so maybe it's not that big problem

**Review by rakhmets — APPROVED** [review]

**Review by yosefe — APPROVED** [review]

**Pull request merged**